crates/re_build_tools/src/hashing.rs

```diff
@@ -0,0 +1,159 @@
+use std::fmt::Write;
+use std::path::{Path, PathBuf};
+use std::{fs, io};
+
+use anyhow::Context as _;
+use sha2::{Digest, Sha256};
+
+// ---
+
+fn encode_hex(bytes: &[u8]) -> String {
+    let mut s = String::with_capacity(bytes.len() * 2);
+    for &b in bytes {
+        write!(&mut s, "{b:02x}").unwrap();
+    }
+    s
+}
+
+/// Walks the directory at `path` in filename order.
+///
+/// If `extensions` is specified, only files with the right extensions will be iterated.
+/// Specified extensions should include the dot, e.g. `.fbs`.
+pub fn iter_dir<'a>(
+    path: impl AsRef<Path>,
+    extensions: Option<&'a [&'a str]>,
+) -> impl Iterator<Item = PathBuf> + 'a {
+    fn filter(entry: &walkdir::DirEntry, extensions: Option<&[&str]>) -> bool {
+        let is_dir = entry.file_type().is_dir();
+        let is_interesting = extensions.map_or(true, |extensions| {
+            extensions.iter().any(|ext| {
+                entry
+                    .file_name()
+                    .to_str()
+                    .map_or(false, |s| s.ends_with(ext))
+            })
+        });
+        is_dir || is_interesting
+    }
+
+    let path = path.as_ref();
+    walkdir::WalkDir::new(path)
+        .sort_by_file_name()
+        .into_iter()
+        .filter_entry(move |entry| filter(entry, extensions))
+        .filter_map(|entry| entry.ok())
+        .filter_map(|entry| entry.file_type().is_file().then(|| entry.into_path()))
+}
+
+/// Given a file path, computes the sha256 hash of its contents and returns an hexadecimal string
+/// for it.
+///
+/// Panics if the file doesn't exist.
+pub fn compute_file_hash(path: impl AsRef<Path>) -> String {
```

Comment:
being in build tools I wonder if this should immediately also track the file by outputting the magic string to rerun on change. What speaks against that?